File: docmindui-python-next/frontend/src/features/chat/types/agent-activity.ts
export type AgentActivityStep = {
  phase: string
  label: string
  detail?: string | null
}

export type KnowledgeSourceTag = {
  document_id: string
  titulo: string
  palavras_chave?: string
  snippet?: string
}

export type AgentActivityMeta = {
  intent: string
  steps: AgentActivityStep[]
  tools: string[]
  sources: KnowledgeSourceTag[]
  documents_in_base: number
  knowledge_base_id?: string | null
  knowledge_base_name?: string | null
  search_label?: string | null
  status_message?: string | null
  reasoning_log?: string | null
}

export const INTENT_LABELS: Record<string, string> = {
  profile: "Perfil do usuário",
  address: "Endereço",
  knowledge: "Base de conhecimento",
  rag: "Documentos (RAG)",
  vision: "Visão + documentos",
  general: "Conversa geral",
}

File: docmindui-python-next/backend/src/schemas/agent_activity.py
from pydantic import BaseModel, Field


class KnowledgeSourceDTO(BaseModel):
    document_id: str
    titulo: str
    palavras_chave: str = ""
    snippet: str = ""


class AgentActivityStepDTO(BaseModel):
    phase: str
    label: str
    detail: str | None = None


class AgentActivityDTO(BaseModel):
    intent: str = ""
    steps: list[AgentActivityStepDTO] = Field(default_factory=list)
    tools: list[str] = Field(default_factory=list)
    sources: list[KnowledgeSourceDTO] = Field(default_factory=list)
    documents_in_base: int = 0
    knowledge_base_id: str | None = None
    knowledge_base_name: str | None = None
    search_label: str | None = None
    status_message: str | None = None
    reasoning_log: str | None = None
